horreum-java-client/src/test/java/io/hyperfoil/tools/HorreumTestBase.java

```diff
@@ -0,0 +1,161 @@
+
+import static org.junit.Assert.assertNotNull;
+
+public class HorreumTestBase {
+
+    static Properties configProperties;
+    public static final String HORREUM_KEYCLOAK_REALM;
+    public static final String HORREUM_KEYCLOAK_BASE_URL;
+    private static final String HORREUM_BASE_URL;
+    public static final String HORREUM_CLIENT_ID;
+
+    public static final String HORREUM_USERNAME;
+    public static final String HORREUM_PASSWORD;
+    private static final String HORREUM_CLIENT_SECRET;
+
+    protected static boolean START_HORREUM_INFRA;
+    protected static boolean CREATE_HORREUM_TEST;
+    protected static boolean HORREUM_DUMP_LOGS;
+
+    protected static HorreumClient horreumClient;
+
+    protected static Test newTest;
+
+    private static Logger log = Logger.getLogger(HorreumTestBase.class);
+
+    static {
+        configProperties = new Properties();
+        InputStream propertyStream = Thread.currentThread().getContextClassLoader().getResourceAsStream("env.properties"); //TODO: make configurable
+        try {
+            if (propertyStream != null) {
+                configProperties.load(propertyStream);
+
+                HORREUM_KEYCLOAK_BASE_URL = getProperty("horreum.keycloak.base-url");
+                HORREUM_BASE_URL = getProperty("horreum.base-url");
+                HORREUM_CLIENT_ID = getProperty("horreum.client-id");
+                HORREUM_KEYCLOAK_REALM = getProperty("keycloak.realm");
+
+                HORREUM_USERNAME = getProperty("horreum.username");
+                HORREUM_PASSWORD = getProperty("horreum.password");
+                HORREUM_CLIENT_SECRET = getProperty("horreum.client_secret");
+
+                START_HORREUM_INFRA = Boolean.valueOf(getProperty("horreum.start-infra"));
+                HORREUM_DUMP_LOGS = Boolean.valueOf(getProperty("horreum.dump-logs"));
+                CREATE_HORREUM_TEST = Boolean.valueOf(getProperty("horreum.create-test"));
+            } else {
+                throw new RuntimeException("Could not load test configuration");
+            }
+        } catch (IOException ioException) {
+            throw new RuntimeException("Failed to load configuration properties");
+        }
+    }
+
+    public static DockerComposeContainer environment = null;
+
+    protected io.hyperfoil.tools.horreum.entity.json.Test getExistingtest(){
+         return horreumClient.testService.get(newTest.id, null);
+    }
+
+    protected static String getProperty(String propertyName) {
+        return configProperties.getProperty(propertyName).trim();
+    }
+
+    @BeforeAll
+    public static void beforeClass() throws Exception {
+        initialiseRestClients();
+
+        log.info("Starting Infra: " + START_HORREUM_INFRA);
+        if (START_HORREUM_INFRA) {
+
+            environment = new DockerComposeContainer(new File("src/test/resources/testcontainers/docker-compose.yml"))
+                    .withLocalCompose(true)
+                    .withExposedService("postgres_1", 5432)
```

Comment:
Also the names of services/containers could be prefixed by `test-`.

Reply:
I can update the port numbers to use ports that will not conflict

iirc, testcontainers prefixes the container name with "testcontainer-{UID}-postrges_1" or something similar to avoid naming conflicts, but will dble check this behaviour